tools/libfrencutils/create_xgrid_util.c

```diff
@@ -47,6 +47,15 @@ int line_intersect_2D_3D(double *a1, double *a2, double *q1, double *q2, double
 void malloc_minmaxavg_lists(const int n, Minmaxavglists *minmaxavglists)
 {
 
+  if( !minmaxavglists->lon_min_list ) free(minmaxavglists->lon_min_list);
```

Comment:
This may be backwards. Perhaps it should be like 
```if(  minmaxavglists->lon_min_list ) free(minmaxavglists->lon_min_list);```
I.e. the ```!``` should not be there? Or perhaps better to compare to NULL? Like:
```if( minmaxavglists->lon_min_list != NULL ) free(minmaxavglists->lon_min_list);``` ?

Reply:
Made this more explicit